kie-maven-plugin/src/main/java/org/kie/maven/plugin/GeneratePMMLModelMojo.java

```diff
@@ -0,0 +1,322 @@
+                    .filter(f -> f.endsWith("drl"))
+                    .collect(Collectors.toSet());
+
+            getLog().info(String.format("Found %d generated files in Canonical Model", generatedFiles.size()));
+
+            MemoryFileSystem mfs = kieModule instanceof CanonicalKieModule ?
+                    ((MemoryKieModule) ((CanonicalKieModule) kieModule).getInternalKieModule()).getMemoryFileSystem() :
+                    ((MemoryKieModule) kieModule).getMemoryFileSystem();
+
+            final String droolsModelCompilerPath = "/generated-sources/drools-model-compiler/main/java";
+            final String newCompileSourceRoot = targetDirectory.getPath() + droolsModelCompilerPath;
+            project.addCompileSourceRoot(newCompileSourceRoot);
+
+            for (String generatedFile : generatedFiles) {
+                final MemoryFile f = (MemoryFile) mfs.getFile(generatedFile);
+                final Path newFile = Paths.get(targetDirectory.getPath(),
+                                               droolsModelCompilerPath,
+                                               f.getPath().toPortableString());
+
+                try {
+                    Files.deleteIfExists(newFile);
+                    Files.createDirectories(newFile.getParent());
+                    Files.copy(f.getContents(), newFile, StandardCopyOption.REPLACE_EXISTING);
+
+                    getLog().info("Generating " + newFile);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    throw new MojoExecutionException("Unable to write file", e);
+                }
+            }
+
+            // copy the META-INF packages file
+            final String path = CanonicalKieModule.getModelFileWithGAV(kieModule.getReleaseId());
+            final MemoryFile packagesMemoryFile = (MemoryFile) mfs.getFile(path);
+            final String packagesMemoryFilePath = packagesMemoryFile.getFolder().getPath().toPortableString();
+            final Path packagesDestinationPath = Paths.get(targetDirectory.getPath(), CLASSES, packagesMemoryFilePath, packagesMemoryFile.getName());
+
+            try {
+                if (!Files.exists(packagesDestinationPath)) {
+                    Files.createDirectories(packagesDestinationPath.getParent());
+                }
+                Files.copy(packagesMemoryFile.getContents(), packagesDestinationPath, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new MojoExecutionException("Unable to write file", e);
+            }
+
+            if (ExecModelMode.shouldDeleteFile(generatePMMLModel)) {
+                deleteDrlFiles(drlFiles);
+            }
+
+        } finally {
+            Thread.currentThread().setContextClassLoader(contextClassLoader);
+        }
+        // REMOVING FLAG TO KNOW THAT BUILD HAS BEEN LAUNCHED BY MAVEN
+        properties.remove("kie-maven-plugin-launcher");
```

Comment:
What about move this line inside `finally` block?

Reply:
@danielezonca 
Removed